packages/element/src/create-interpolate-element.js

```diff
@@ -0,0 +1,235 @@
+		case 'children':
+			pattern = escapeRegExp( openTag ) + '(.*)' + escapeRegExp( closeTag );
+			break;
+		case 'propValue':
+			pattern = escapeRegExp( searchString );
+			break;
+		default:
+			pattern = escapeRegExp( selfClosingTag );
+	}
+	return new RegExp( pattern );
+};
+
+/**
+ * Generates and returns the regular expression used for splitting a string
+ *
+ * @param {string} searchString  The search string serving as the base for the
+ *                               expression.
+ *
+ * @return {RegExp}  The generated regular expression
+ */
+const getSplitRegEx = ( searchString ) => new RegExp( '(' + escapeRegExp( searchString ) + ')' );
+
+/**
+ * Generates a String.prototype.match value for the incoming arguments.
+ *
+ * @param {string} interpolatedString  The string the match is performed on.
+ * @param {string} searchString				 The string used as the base for the
+ *                                     search
+ * @param {Object} conversionConfig    The configuration object for the
+ *                                     interpolation being performed.
+ *
+ * @return {Array|null}	An array if there is a match or null if not.
+ */
+const getMatchFromString = (
+	interpolatedString,
+	searchString,
+	conversionConfig
+) => {
+	const tags = getTagsFromString( searchString );
+	// first try children reg ex.  If there is a match, then return.
+	const match = interpolatedString.match(
+		getRegEx( tags, searchString, 'children' )
+	);
+	if ( match !== null ) {
+		conversionConfig.hasChildren = true;
+		return match;
+	}
+	// no children, try matching on another pattern and return.
+	const regEx = getHasPropValue( conversionConfig ) ?
+		getRegEx( tags, searchString, 'propValue' ) :
+		getRegEx( tags, searchString );
+	return interpolatedString.match( regEx );
+};
+
+// index for keys
+// This is external to `recursiveCreateElement` and reset in
+// `createInterpolateElement` because of the recursion.
+let keyIndex = -1;
+
+/**
+ * Used to recursively create elements from the interpolation string using the
+ * conversion map.
+ *
+ * @param {string}  potentialElement  The interpolation string (or fragment)
+ *                                    being processed.
+ * @param {Array[]} conversionMap     The interpolation map used for converting
+ *                                    the string to a react element.
+ *
+ * @return {Element|string|Array}  A react element, string or array.
+ */
+const recursiveCreateElement = ( potentialElement, conversionMap ) => {
+	/**
+	 * If the conversion map is not a valid array or empty then just return the
+	 * element.
+	 */
+	if ( ! Array.isArray( conversionMap ) || ! conversionMap.length ) {
+		return potentialElement;
+	}
+	const [ mapItem ] = conversionMap.slice( 0, 1 );
+	const [ searchString, conversionConfig ] = mapItem;
+
+	/**
+	 * This short circuits the process if the conversion map has an invalid config.
+	 */
+	if ( ! searchString || ! conversionConfig ) {
+		return potentialElement;
+	}
```

Comment:
I'm wondering if this sort of validation isn't the kind of task for the `createInterpolateElement` wrapper, since we can check the whole map beforehand in one pass.

Reply:
I feel that will create an additional unnecessary iteration step.  Unless I'm not understanding how what you would propose would happen?  Keep in mind `searchString` and `conversionConfig` have been extracted from the current iterator position.